src/IndexingWrapper.cpp

```diff
@@ -243,6 +245,71 @@ indexWrapper(Rcpp::CharacterVector & dbpath, Rcpp::DataFrame & dataFrame, Rcpp::
 	    }
 	}
 
+	//Add value slots
+	for (Rcpp::List::iterator it = slotList.begin(); it != slotList.end(); ++it) {
+	    Rcpp::List list = it.operator*();
+	    bool serialiseVal = false;
+	    bool dfVal = true; // use original row values in data frame
+	    int slot;
+	    std::string val;
+
+	    try {
+		slot = Rcpp::as<int>(list["slot"]);
+	    } catch (...) {
+		Rcpp::stop("Slot number should be specified.");
+	    }
+
+	    if (list.containsElementNamed("serialise")) serialiseVal = Rcpp::as<bool>(list["serialise"]);
+	    if (list.containsElementNamed("values")) dfVal = false; // use a column vector of modified row values
+
+	    if (dfVal) {
+		int colNo;
+
+		try {
+		    if (list.containsElementNamed("index")) {
+			colNo = Rcpp::as<int>(list["index"]);
+		    } else {
+			std::string colName = Rcpp::as<std::string>(list["name"]);
+			colNo = colIndex(df, colName);
+		    }
+		} catch (...) {
+		    Rcpp::stop("Either name or index of the columns should be specified.");
+		}
+
+		if (colNo < 0 || colNo > dfCols) Rcpp::stop("Invalid argument for valueSlots");
+
+		const string & field = Rcpp::as<std::string>(dfRow[colNo]);
+
+		if (serialiseVal) {
+		    double dVal;
+		    stringstream ss(field);
+		    ss >> dVal;
+		    doc.add_value(slot, Xapian::sortable_serialise(dVal));
+		} else {
+		    doc.add_value(slot, field);
+		}
+	    } else {
+		std::string vecType = Rcpp::as<std::string>(list["type"]);
+		if (vecType == "double") {
+		    Rcpp::DoubleVector dVec = list["values"];
+		    double dVal = dVec[i];
+		    if (!(Rcpp::DoubleVector::is_na(dVal))) doc.add_value(slot, Xapian::sortable_serialise(dVal));
+		}
+		if (vecType == "character") {
+		    Rcpp::StringVector sVec = list["values"];
+		    std::string sVal = Rcpp::as<std::string>(sVec[i]);
+		    if (serialiseVal) {
+			double dVal;
+			stringstream ss(sVal); //turn field into a stream
```

Comment:
Since you didn't have this comment when doing the same thing earlier on, you shouldn't need it here.